tests/test_notify_v2.py

```diff
@@ -0,0 +1,415 @@
+        cls.s3 = boto3.client('s3')
+
+    @classmethod
+    def tearDownClass(cls):
+        for replica in Replica:
+            subs = [s for s in subscriptions_v2.get_subscriptions_for_owner(replica, cls.owner)
+                    if s['owner'] == cls.owner]
+            for s in subs:
+                subscriptions_v2.delete_subscription(replica, cls.owner, s['uuid'])
+
+    @testmode.integration
+    def test_versioned_tombstone_notifications(self, replica=Replica.aws):
+        bucket = get_env('DSS_S3_BUCKET_TEST')
+        notification_object_key = f"notification-v2/{uuid4()}"
+        url = self.s3.generate_presigned_url(
+            ClientMethod='put_object',
+            Params=dict(Bucket=bucket, Key=notification_object_key, ContentType="application/json")
+        )
+        subscription = self._put_subscription(
+            {
+                'callback_url': url,
+                'method': "PUT",
+                'jmespath_query': "admin_deleted==`true`"
+            },
+            replica
+        )
+        bundle_uuid, bundle_version = self._upload_bundle(replica)
+        self._tombstone_bundle(replica, bundle_uuid, bundle_version)
+
+        notification = self._get_notification_from_s3_object(bucket, notification_object_key)
+        self.assertEquals(notification['subscription_id'], subscription['uuid'])
+        self.assertEquals(notification['match']['bundle_uuid'], bundle_uuid)
+        self.assertEquals(notification['match']['bundle_version'], f"{bundle_version}.dead")
+
+    @testmode.integration
+    def test_unversioned_tombstone_notifications(self, replica=Replica.aws):
+        bucket = get_env('DSS_S3_BUCKET_TEST')
+        notification_object_key = f"notification-v2/{uuid4()}"
+        url = self.s3.generate_presigned_url(
+            ClientMethod='put_object',
+            Params=dict(Bucket=bucket, Key=notification_object_key, ContentType="application/json")
+        )
+        subscription = self._put_subscription(
+            {
+                'callback_url': url,
+                'method': "PUT",
+                'jmespath_query': "admin_deleted==`true`"
+            },
+            replica
+        )
+        bundle_uuid, bundle_version = self._upload_bundle(replica)
+        bundle_uuid, bundle_version = self._upload_bundle(replica, bundle_uuid)
+        self._tombstone_bundle(replica, bundle_uuid)
+
+        notification = self._get_notification_from_s3_object(bucket, notification_object_key)
+        self.assertEquals(notification['subscription_id'], subscription['uuid'])
+        self.assertEquals(notification['match']['bundle_uuid'], bundle_uuid)
+        # TODO:
+        # Multiple notifications should be delivered for this test. However, the notification
+        # test infrastructure (presigned s3 url) cannot track multiple deliveries. Need another
+        # mechanism.
+        # Brian Hannafious 2019-01-30
+
+    @testmode.integration
+    def test_queue_notification(self):
```

Comment:
Would it be good to assert that `self._get_notification_from_s3_object(bucket, key)` doesn't return a `notification['subscription_id'] == subscription['uuid']` at the start of the test?

Reply:
`self._get_notification_from_s3_object(bucket, key)` looks for an object whose key contains a uuid generated at the beginning of this test. If that key already exists in the bucket then we are cosmically unlucky.